sw/device/lib/dif/dif_i2c.h

```diff
@@ -186,6 +204,33 @@ typedef struct dif_i2c_fmt_flags {
   bool suppress_nak_irq;
 } dif_i2c_fmt_flags_t;
 
+/**
+ * The I2C Target device records the following signals with recieved data
+ */
+typedef enum dif_i2c_signal{
+  /**
+   * The associated byte was recieved with a START signal, and should be the
+   * matching address and R/W bit
+   */
+  kDifI2cSignalStart,
+  /**
+   * The associated byte was recieved after a STOP signal, so the transaction is
+   * over and the byte is junk
+   */
+  kDifI2cSignalStop,
+  /**
+   * The associated byte was recieved with a repeated START signal and should
+   * probably be handled as the address for a subsequent transaction (it  may
+   * match)
```

Comment:
Haha, this seems a little passive-aggressive. We should figure this out and make the docs more confident-sounding. ;)